packages/timeline/src/common/timeline-model.ts

```diff
@@ -0,0 +1,80 @@
+/********************************************************************************
+ * Copyright (C) 2020 RedHat and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+// some code copied and modified from https://github.com/microsoft/vscode/blob/3aab025eaebde6c9544293b6c7554f3f583e15d0/src/vs/workbench/contrib/timeline/common/timeline.ts
+
+import { Command, Disposable, Event } from '@theia/core/lib/common';
+import URI from '@theia/core/lib/common/uri';
+
+export class TimelineItem {
```

Comment:
there are some classes which will be pulled in the plugin host process unnecessary. Could you separate this file on model and protocol. In protocol keep types used in the plugin host process, model should be moved to browser and used only there.

Reply:
or convert everything to interfaces that this file does not contribute anything at runtime

Reply:
Converted the `TimelineItem` to interface, so only interfaces are present here.